UserTools/test_vertices/test_vertices.cpp

```diff
@@ -54,21 +141,35 @@ bool test_vertices::Initialise(std::string configfile, DataModel &data){
 bool test_vertices::Execute(){
 
 
-  //do stuff with m_data->Samples
-  /*
-  for( std::vector<SubSample>::const_iterator is=m_data->Samples.begin(); is!=m_data->Samples.end(); ++is){
-    PMTids.push_back(is->m_PMTid);
-    times.push_back(is->m_time);
-  }
-  */
-  int the_output;
-  //  the_output = CUDAFunction(m_data->Samples.at(0).m_PMTid, m_data->Samples.at(0).m_time);
+  std::vector<SubSample> & samples = m_data->IDSamples;
+
+  for( std::vector<SubSample>::const_iterator is=samples.begin(); is!=samples.end(); ++is){
 #ifdef GPU   
-  the_output =   GPU_daq::test_vertices_execute();
+
+    std::vector<int> trigger_ns;
+    std::vector<int> trigger_ts;
+    m_time_int.clear();
+    for(unsigned int i = 0; i < is->m_time.size(); i++) {
+      m_time_int.push_back(is->m_time[i]);
+    }
+    GPU_daq::test_vertices_execute(is->m_PMTid, m_time_int, &trigger_ns, &trigger_ts);
+    for(int i=0; i<trigger_ns.size(); i++){
+      m_data->IDTriggers.AddTrigger(kTriggerUndefined,
+				    TimeDelta(trigger_ts[i]) + m_trigger_gate_down + is->m_timestamp, 
+				    TimeDelta(trigger_ts[i]) + m_trigger_gate_up + is->m_timestamp,
+				    TimeDelta(trigger_ts[i]) + is->m_timestamp,
+				    std::vector<float>(1, trigger_ns[i]));
```

Comment:
`m_trigger_gate_X` are not `TimeDelta` instances, so they should be converted explicitly to avoid confusion. We can just add them to the other times before turning them into a `TimeDelta` together.

```suggestion
      m_data->IDTriggers.AddTrigger(kTriggerUndefined,
				    TimeDelta(trigger_ts[i] + m_trigger_gate_down) + is->m_timestamp, 
				    TimeDelta(trigger_ts[i] + m_trigger_gate_up) + is->m_timestamp,
				    TimeDelta(trigger_ts[i]) + is->m_timestamp,
				    std::vector<float>(1, trigger_ns[i]));
```

(I really hope I did not tell you exactly the opposite thing in a previous review comment...)

Reply:
done